### DISCUSSION TITLE
How to add my custom additions from emacs

### DISCUSSION THREAD

**Discussion by Odisej75** [syl20bnr/spacemacs#15855] [Q&A] [answered]
Ok, I know you will hate me for it... but I am a bit lost with spacemacs (although it looks great). I am used to emacs  where I added some changes mainly when writing text (that's what I mostly do). Where can I add those lines so that spacemacs would work the same when i open a TXT file? First change that I use is spellchecking, second is setting the edges of the text by pressing F5, third is havin a character count on the mode line.

The "hooks" used are:
```
(add-hook 'text-mode-hook 'wc-mode)
(add-hook 'text-mode-hook 'visual-line-mode)
(dolist (hook '(text-mode-hook))
        (add-hook hook (lambda () (flyspell-mode 1))))
(defun my-set-margins ()
  "Set margins in current buffer."
  (setq left-margin-width 1)
  (setq right-margin-width 1))

(add-hook 'text-mode-hook 'my-set-margins)

(defun my-toggle-margins ()
"Set margins in current buffer."
(interactive)
  (if (or (> left-margin-width 0) (> right-margin-width 0))
    (progn
      (setq left-margin-width 0)
      (setq right-margin-width 0)
      (set-window-buffer (selected-window) (current-buffer)))
    (setq left-margin-width 26)
    (setq right-margin-width 26)
    (set-window-buffer (selected-window) (current-buffer))))

(global-set-key [f5] 'my-toggle-margins)
```

Would be happy if somebody would at least direct me on the correct path.

**Comment by pataquets** [comment, marked as answer]
[Spacemacs' spell checking layer](https://develop.spacemacs.org/layers/LAYERS.html#spell-checking) might be useful to achieve it, read the layer configuration docs. Also, check out the other layers for assembling your config.
Tip: using Markdown's [syntax highlighting in code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) will make your posted code easier to read.
